scripts/ci/gitlab/pipeline/test.yml

```diff
@@ -72,3 +76,31 @@ test-deterministic-wasm:
     - .compiler-info
   script:
     - ./scripts/ci/gitlab/test_deterministic_wasm.sh
+
+cargo-fmt:
+  stage:                           test
+  variables:
+    RUSTY_CACHIER_TOOLCHAIN: nightly
+  extends:
+    - .docker-env
+    - .test-refs
+  script:
+    # - rusty-cachier snapshot create
+    - cargo +nightly fmt --all -- --check
+    # - rusty-cachier cache upload
+
+cargo-clippy:
+  stage:                           test
```

Comment:
@alvicsam 
First, why `cargo-fmt` is in `check.yml` in polkadot and `test.yml` in substrate?
Seconds, for this "pipeline optimization using GitLab's DAGs, do we really need other jobs depending on `cargo-clippy`? What does this optimize exactly? Cancellation of jobs if clippy fails? We probably don't want to cancel other jobs in this case.

Reply:
1. For historical reasons I guess? 
2. DAG is needed to skip stages. For example in polkadot one of the longest is `test-linux-stable` (it takes about 30 min). Using DAG allows to run all other jobs in different stages without waiting `test-linux-stable`. Without DAG you would wait only `test` stage for 30 minutes and then comes `build` , and then publish etc. In this PR needs are debatable and I'll adjust them later

Reply:
I think what we want is  `cargo-clippy` to be run in parallel to other checks but allow all other stages to run even if `cargo-clippy` job fails. Which stage should be it be added in this case and what dependencies should be specified?

Reply:
Added DAG to the clippy job. It will run always but won't interrupt other jobs even if fails